src/cheri_insts_cext.sail

```diff
@@ -0,0 +1,170 @@
+/*  programme (grant agreement 789108, ELVER).                                           */
+/*                                                                                       */
+/*  Redistribution and use in source and binary forms, with or without                   */
+/*  modification, are permitted provided that the following conditions                   */
+/*  are met:                                                                             */
+/*  1. Redistributions of source code must retain the above copyright                    */
+/*     notice, this list of conditions and the following disclaimer.                     */
+/*  2. Redistributions in binary form must reproduce the above copyright                 */
+/*     notice, this list of conditions and the following disclaimer in                   */
+/*     the documentation and/or other materials provided with the                        */
+/*     distribution.                                                                     */
+/*                                                                                       */
+/*  THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS''                   */
+/*  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED                    */
+/*  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A                      */
+/*  PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR                  */
+/*  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,                         */
+/*  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT                     */
+/*  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF                     */
+/*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND                  */
+/*  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,                   */
+/*  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT                   */
+/*  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF                   */
+/*  SUCH DAMAGE.                                                                         */
+/*=======================================================================================*/
+
+/* Compressed capability loads/stores (encodings differ between RV32/RV64) */
+union clause ast = C_CLC : (bits(9), cregidx, cregidx)
+
+mapping clause encdec_compressed_capmode = C_CLC(ui8 @ ui76 @ ui54 @ 0b0000, cs1c, cdc) if sizeof(xlen) == 64
+  <-> 0b001 @ ui54 : bits(2) @ ui8 : bits(1) @ cs1c : cregidx @ ui76 : bits(2) @ cdc : cregidx @ 0b00 if sizeof(xlen) == 64
+mapping clause encdec_compressed_capmode = C_CLC(0b0 @ ui76 @ ui53 @ 0b000, cs1c, cdc) if sizeof(xlen) == 32
+  <-> 0b011 @ ui53 : bits(3) @ cs1c : cregidx @ ui76 : bits(2) @ cdc : cregidx @ 0b00 if sizeof(xlen) == 32
+mapping clause assembly = C_CLC(uimm, cdc, csc1)
```

Comment:
`cdc` and `csc1` swapped here vs decode clauses. Same in assembly clause.

Also inconsistent use of `cs1c` vs `csc1`.

Reply:
This fix works but means the order of the operands is now 'middle-endian' matching neither the encoding or the `LoadCapImm` constructor. I think it would be slightly less confusing to use the same order as `LoadCapImm`.

Reply:
Well, I suppose it matches the encoding as far as it can it's just unfortunately that `LoadCapImm` chose a different order.